test-js/unit/pentaho/lang/Base.Spec.js

```diff
@@ -789,202 +816,206 @@ define([
-                    var cat_descriptor = O.getPropertyDescriptor(Cat.prototype, "size");
+              expectIt({a: 1, b: 2});
+              expectIt({b: 2, a: 1});
+            });
 
-                    expect(persa_descriptor.get).not.toBe(cat_descriptor.get);
-                    expect(persa_descriptor.set).toBe(cat_descriptor.set);
-                  });
+            it("should take effect in implement calls of the new class", function() {
 
-                  it("only setter", function() {
-                    var persa_descriptor = O.getPropertyDescriptor(Persa.prototype, "height");
-                    var cat_descriptor = O.getPropertyDescriptor(Cat.prototype, "height");
+              function expectIt(spec) {
+                var index = 1, aIndex, bIndex;
 
-                    expect(persa_descriptor.get).toBe(cat_descriptor.get);
-                    expect(persa_descriptor.set).not.toBe(cat_descriptor.set);
-                  });
+                var Derived = Base.extend({
+                  extend_order: ["a", "b"],
+                  get a()  { return this._a; },
+                  set a(v) {
+                    aIndex = index++;
+                    this._a = v;
+                  },
+
+                  get b()  { return this._b; },
+                  set b(v) {
+                    bIndex = index++;
+                    this._b = v;
+                  }
                 });
-              });
 
-              describe("overridden methods", function() {
-                describe("instance level", function() {
-                  var persa_cat;
+                Derived.implement(spec);
 
-                  beforeEach(function() {
-                    persa_cat = new Persa();
-                  });
+                expect(Derived.prototype.a).toBe(1);
+                expect(Derived.prototype.b).toBe(2);
 
-                  it("should be able to call its base method", function() {
-                    persa_cat.drink("orange juice");
+                expect(aIndex).toBe(1);
+                expect(bIndex).toBe(2);
+              }
 
-                    expect(persa_cat.drank.length).toBe(1);
-                    expect(persa_cat.drank[0]).toBe("orange juice with sugar");
-                  });
+              expectIt({a: 1, b: 2});
+              expectIt({b: 2, a: 1});
+            });
 
-                  it("getters/setters should be able to call its base method", function() {
-                    persa_cat.color = "black";
+            it("should add to the ordered props of the base class", function() {
 
-                    expect(persa_cat.color).toBe("bald blackish");
-                  });
+              function expectIt(spec) {
+                var index = 1, aIndex, bIndex, cIndex;
 
-                  it("should be able to call its base method even if there isn't one", function() {
-                    expect(persa_cat.run()).toBeUndefined();
-                  });
+                var Derived1 = Base.extend({
```

Comment:
The base order wins. Duplicate elements are only processed in their first occurrence